### PR TITLE
Added graying out for UI options as in #3755

### PR DESCRIPTION
**Pull request by lasalvavida**
Add the changes suggested in #3755 and made the pull request to the correct branch this time.

### CODE REVIEW

**Comment by lilleyse** [comment]
I haven't looked too much into it, but you may be able to utilize knockout to achieve something cleaner. If there's no obvious solution, I'm fine with what you have here.

**Comment by lasalvavida** [comment]
I'll look into it

**Comment by lasalvavida** [comment]
Updated

File: Apps/Sandcastle/gallery/development/Shadows.html [hunk 1]
```diff
@@ -144,6 +167,7 @@
 Cesium.knockout.getObservable(viewModel, 'cascades').subscribe(updateShadows);
 Cesium.knockout.getObservable(viewModel, 'lightSource').subscribe(updateShadows);
 Cesium.knockout.getObservable(viewModel, 'size').subscribe(updateSettings);
+Cesium.knockout.getObservable(viewModel, 'softShadows').subscribe(updateShadows);
```
Comment (lilleyse) [inline Apps/Sandcastle/gallery/development/Shadows.html]: The latest shadows code that you merged in also added this line above, so you can remove this.

File: Apps/Sandcastle/gallery/development/Shadows.html [hunk 2]
```diff
@@ -109,23 +113,42 @@
...
     softShadows : false,
+    softShadowsEnabled : true,
     cascadeOptions : [1, 4],
     cascades : 4,
+    cascadesEnabled : true,
     lightSourceOptions : ['Freeform', 'Sun', 'Fixed', 'Point'],
     lightSource : 'Freeform',
     sizeOptions : [256, 512, 1024, 2048],
-    size : 1024
+    size : 1024,
```
Comment (lilleyse) [inline Apps/Sandcastle/gallery/development/Shadows.html]: Remove comma.

**Comment by lilleyse** [comment]
The dropdowns and check-boxes are still interactive when they should be disabled.

File: Apps/Sandcastle/gallery/development/Shadows.html [hunk 3]
```diff
@@ -215,6 +239,18 @@
...
     updateSettings();
+    updateUI();
+}
+
+function updateUI() {
+    uiOptions.all.forEach(function(setting) {
+        if (uiOptions.disable[viewModel.lightSource].indexOf(setting) > -1) {
+            viewModel[setting + 'Enabled'] = false;
+        }
+        else {
```
Comment (lilleyse) [inline Apps/Sandcastle/gallery/development/Shadows.html]: Small thing, but we usually put the else on the same line.

**Comment by lilleyse** [comment]
That's all my comments. I like the new approach.

**Comment by lasalvavida** [comment]
Updated

**Comment by lilleyse** [comment]
Awesome this looks great!

**Pull request merged**